src/backend/tcop/postgres.c

```
@@ -4850,8 +4844,32 @@ PostgresMain(int argc, char *argv[],
 					if (seqServerHostlen > 0)
 						seqServerHost = pq_getmsgbytes(&input_message, seqServerHostlen);
 
+					numSlices = pq_getmsgint(&input_message, 4);
```

I think it's better to do it in one loop and error out if 'localslice' is not found